eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h

```diff
@@ -19,16 +18,26 @@ namespace vespalib::tensor {
 class SparseTensorAddressBuilder
 {
 private:
-    std::vector<char> _address;
+    vespalib::Array<char> _address;
 
+protected:
     void append(vespalib::stringref str) {
-        const char *cstr = str.c_str();
-        _address.insert(_address.end(), cstr, cstr + str.size() + 1);
+        for (size_t i(0); i < str.size() + 1; i++) {
+            _address.push_back_fast(str[i]);
```

Comment:
I do not think accessing the 0-terminator like this is defined.

Reply:
also, better would be to insert explicit 0 after the string, to also support appending substrings...

Reply:
I agree, just made a minimal change. I will fix it at next crossroads.